content/doc/book/pipeline/syntax.adoc

```diff
@@ -0,0 +1,851 @@
+skipDefaultCheckout:: Skip checking out code from source control by default in
+the `agent` directive. For example: `options { skipDefaultCheckout() }`
+timeout:: Set a timeout period for the Pipeline run, after which Jenkins should
+abort the Pipeline. For example: `options { timeout(time: 1, unit: 'HOUR') }`
+retry:: On failure, retry the entire Pipeline the specified number of times.
+For example: `options { retry(3) }`
+timestamps:: Prepend all console output generated by the Pipelien run with the
+time at which the line was emitted. For example: `options { timestamps() }`
+
+[[options-example]]
+===== Example
+
+[pipeline]
+----
+// Declarative //
+pipeline {
+    agent any
+    options {
+        timeout(time: 1, unit: 'HOUR') /* <1> */
+    }
+    stages {
+        stage('Example') {
+            steps {
+                echo 'Hello World'
+            }
+        }
+    }
+}
+----
+<1> Specifying a global execution timeout of one hour, after which Jenkins will
+abort the Pipeline run.
+
+[NOTE]
+====
+A comprehensive list of available options is pending the completion of
+link:https://issues.jenkins-ci.org/browse/INFRA-1053[INFRA-1503].
+====
+
+==== parameters
+
+The `parameters` directive provides a list of parameters which a user should
+provide when triggering the Pipeline. The values for these user-specified
+parameters are made available off of the `params` object within the Pipeline,
+see the <<parameters-example>> for its specific usage.
+
+[cols="^10h,>90a",role=syntax]
+|===
+| Required
+| No
+
+| Parameters
+| _None_
+
+| Allowed
+| Only once, inside the `pipeline` block.
+|===
+
+===== Available Parameters
+
+string:: A parameter of a string type, for example: `parameters { string(name: 'DEPLOY_ENV', defaultValue: 'staging', description: '') }`
+booleanParam:: A boolean parameter, for example: `parameters { booleanParam(name: 'DEBUG_BUILD', defaultValue: true, description: '') }`
+
+
+[[parameters-example]]
+===== Example
+
+[pipeline]
+----
+// Declarative //
+pipeline {
+    agent any
+    parameters {
+        string(name: 'PERSON', defaultValue: 'Mr Jenkins', description: 'Who should I say hello to?'
+    }
+    stages {
+        stage('Example') {
+            steps {
+                echo "Hello ${params.PERSON}"
+            }
+        }
+    }
+}
+----
+
+[NOTE]
+====
+A comprehensive list of available parameters is pending the completion of
+link:https://issues.jenkins-ci.org/browse/INFRA-1053[INFRA-1503].
+====
+
+==== triggers
+
+The `triggers` directive defines the automated ways in which the Pipeline
+should be re-triggered. For Pipelines which are integrated with a source such
+as GitHub or BitBucket, `triggers` may not be necessary as webhooks-based
+integration will likely already be present. Currently the only two available
+triggers are `cron` and `pollScm`.
+
+[cols="^10h,>90a",role=syntax]
+|===
+| Required
+| No
+
+| Parameters
+| _None_
+
+| Allowed
+| Only once, inside the `pipeline` block.
```

Comment:
Presumably multiple triggers can be added to the block (e.g. poll the SCM regularly, but build anyway, once per night) — it would be nice to mention that.